Assets/Mirror/Runtime/NetworkReader.cs

```diff
@@ -322,6 +322,10 @@ public static List<T> ReadList<T>(this NetworkReader reader)
         public static T[] ReadArray<T>(this NetworkReader reader)
         {
             int length = reader.ReadInt32();
+            if (reader.Position + length > reader.Length)
```

Comment:
```suggestion
            // this assumes that a reader for T reads atleast 1 bytes
            // we can't know the exact size of T because it could have a user created reader
            if (reader.Position + length > reader.Length)
```